ansible/roles/eos/tasks/ceos_ensure_reachable.yml

```diff
@@ -0,0 +1,49 @@
+- block:
+  - name: wait for container's mgmt-ip reachable
+    wait_for:
+      port: 22
+      host: "{{ ansible_host }}"
+      search_regex: OpenSSH
+      delay: 0
+      timeout: 120
+    register: ip_result
+    ignore_errors: yes
+    delegate_to: "{{ VM_host[0] }}"
+
+  - name: restart cEOS container
+    when: ip_result.failed == true and 'Timeout when waiting' in ip_result.msg
```

Comment:
The common practice is to put `when` to the end of a task.
Just checking `ip_result.failed` is enough?

Reply:
1. moved when to the end of the task
2. I would like to restrain restart only in time out cases, but the msg assertion may dismatch in different language, refined the logic with property elapsed, it's the time that last wait_for task cost, if elapsed >= timeout_threshold, that indicates a timeout case.